tests/data_context/test_data_context_datasources.py

```diff
@@ -453,3 +453,34 @@ def test_list_datasources() -> None:
     context.sources.add_pandas(datasource_name)
 
     assert len(context.list_datasources()) == 2
+
+
+@pytest.mark.unit
+def test_get_available_data_assets_names() -> None:
+    project_config = DataContextConfig(
+        store_backend_defaults=InMemoryStoreBackendDefaults()
+    )
+    project_config.datasources = {
+        "my_datasource_name": {
+            "class_name": "Datasource",
+            "data_connectors": {},
+            "execution_engine": {
+                "class_name": "PandasExecutionEngine",
+                "module_name": "great_expectations.execution_engine",
+            },
+            "module_name": "great_expectations.datasource",
+            "data_connectors": {
+                "default_runtime_data_connector_name": {
+                    "class_name": "RuntimeDataConnector",
+                    "module_name": "great_expectations.datasource.data_connector",
+                    "batch_identifiers": ["default_identifier_name"],
+                },
+            },
+        }
+    }
+    context = gx.get_context(project_config=project_config)
+
+    datasource_name = "my_experimental_datasource_awaiting_migration"
+    context.sources.add_pandas(datasource_name)
+
+    assert len(context.get_available_data_asset_names()) == 2
```

Comment:
What happens when you pass `datasource_names`? Do you get the same behavior for the block-config datasource as the fluent datasource?